islandora_solr_config/includes/rss_results.inc

```diff
@@ -191,7 +190,85 @@ class IslandoraSolrResultsRSS extends IslandoraSolrResults {
     $result['url'] = $base_url;
     $result['description'] = t('Aggregated search results of: @query', array('@query' => $query));
     $result['langcode'] = NULL;
+    $feed_url = request_uri();
+    $page_params = drupal_get_query_parameters();
+
+    if (!isset($page_params['page'])) {
+      $page_number = '0';
+    }
+    else {
+      $page_number = $page_params['page'];
+    }
+    if (isset($page_params['page'])) {
+      $first_params = $page_params;
+      unset($first_params['page']);
+      $link_first = url(
+        request_path(),
+        array(
+          'absolute' => TRUE,
+          'query' => $first_params,
+        )
+      );
+    }
+    else {
+      $link_first = $base_url . $feed_url;
+    }
+
+    $page_next = $page_number + 1;
+    if ($page_number > 0) {
+      $page_prev = $page_number - 1;
+    }
+
+    $next_page = array(
+      'query' => array(
+        'page' => $page_next,
+      )+$page_params,
+    );
+
+    // Only set Next link if there is a next page.
+    $rss_limit = variable_get('islandora_solr_config_rss_limit', 50);
+    if (count($docs) == $rss_limit) {
```

Comment:
Is there a better way maybe to get the number of results? Seems like if there are exactly 200 results then this would show there being another page when there isn't actually one.

Page 0 - 50
Page 1 - 50
Page 2 - 50
Page 3 - 50
Page 4 - 0

Reply:
Good point! Managed to fix the total problem -- just a matter of finding the `[numFound]` parameter and dividing by the Solr limit.